### DISCUSSION TITLE
Group description

### DISCUSSION THREAD

**Discussion by bajce** [pnp/pnpcore#772] [General] [answered]
I use following code to create new SharePoint group and set the description:

`var createdGroup = context.Web.SiteGroups.Add(groupName);`
`createdGroup.Description = groupDescription;`
`createdGroup.Update();`

When I open group page on SharePoint the description is not set. Using PnP.PowerShell I can see that description is set. (Description does not contains any HTML and its not longer then 511).

Am I doing something wrong?

I also saw in [documentation](https://pnp.github.io/pnpcore/using-the-sdk/security-groups.html#editing-a-group) that its not allowed to use HTML in description, is there any specific reason for that? PnP.PowerShell allows that for example. That could be useful for Links of Site same as default SharePoint groups have.

**Comment by jansenbe** [comment, marked as answer]
@bajce : thanks for reporting this, I'll give it a go and report back

**Reply by jansenbe** [reply]
@bajce : I've just wrote below test case and for me things work. Are you using the latest version? Please try with below sample code and verify:

```csharp
using (var context = await TestCommon.Instance.GetContextAsync(TestCommon.TestSite))
{
    var groupName = TestCommon.GetPnPSdkTestAssetName("TestGroupWithDescription");
    ISharePointGroup updatedSiteGroup = null;

    try
    {
        var siteGroup = await context.Web.SiteGroups.AddAsync(groupName);

        siteGroup.Description = "This is my custom description";

        await siteGroup.UpdateAsync();

        updatedSiteGroup = await context.Web.SiteGroups.FirstOrDefaultAsync(g => g.Title == groupName);

        Assert.IsTrue(siteGroup.Description == "This is my custom description");
    }
    finally
    {
        if (updatedSiteGroup != null)
        {
            await updatedSiteGroup.DeleteAsync();
        }
    }
}

```

**Reply by bajce** [reply]
@jansenbe: Maybe I didn't explain in correctly. When I update group description property using PnP.Core.SDK, using PnP.Core.SDK or PnP.PowerShell to read the group description property I can verify that the description is set as expected, however I would also expect that new description would be set on group overview page  ('.../sites/SiteName/_layouts/15/groups.aspx') or on edit group page ('.../sites/SiteName/_layouts/15/editgrp.aspx?Group=XXX'). When I update description by using PnP.PowerShell, the changes are visible on both (group overview and group edit) pages.

With my further research I found out that field shown on group overview page is not group description, but "Notes" field of group list item.

```
var item = web.SiteUserInfoList.Items.GetById(group.Id, i => i["Notes"])
item["Notes"] = "This is my custom description";
item.Update();
```
Using this code I was able to update the field I believed it was group description.

I also tested that in on-premises environment and when updating group description using CSOM, both fields (group description and Notes field of group list item) are updated (which is same as when using PnP.PowerShell for SPO)

Im using latest stable version (1.5.0)

**Reply by jansenbe** [reply]
Thanks for the research @bajce , for now I'll add your approach to the documentation, might add an extra "SetGroupDescription" method later on that integrates this

**Reply by jansenbe** [reply]
Here's the code that I'll add to the docs @bajce:

```csharp
await context.Web.EnsurePropertiesAsync(p => p.SiteUserInfoList);
var item = await context.Web.SiteUserInfoList.Items.GetByIdAsync(siteGroup.Id);
item["Notes"] = "This is my custom description";
await item.UpdateAsync();
```

**Reply by bajce** [reply]
@jansenbe great thanks!
